### PR TITLE
CONSOLE-3031: Remove `font-awesome` and `openshift-logos-icon`

### PR DESCRIPTION
**Pull request by logonoff**
<h2 id="feature-fix">CONSOLE Features and Fixes:</h2>

https://issues.redhat.com/browse/CONSOLE-3031

<h2 id="solution-description">Solution description</h2> 

- Removed `font-awesome` 
- Removed `openshift-logos-icon` due to non-usage
- `fa-spin` was changed to a custom `co-spin`, 
- the font-awesome usage in pesudoelement `co-external-link` and `co-goto-arrow` were replaced with mask-images (so that the icons can retain the text color while keeping usage the same), 
- `co-modal-btn-link` was removed due to non-usage across actively maintained dynamic plugins on GitHub
- Font awesome icons used in React were replaced with the PatternFly equivalent
  - if the html element was stored as a string then inserted into the DOM later, a function `getSvgFromPfIconConfig` was written to generate an SVG element from the PatternFly icon exports

<h2 id="test-cases">Test cases:</h2>

- All icons changed should look about the same

<h2 id="screenshots">Screen shots / gifs / design review:</h2>

Corporate needs you to find the differences between this picture and this picture:

Before:
image
image
image
image

After:
image
image
image
image
image

### CODE REVIEW

File: frontend/packages/console-shared/src/components/markdown-extensions/utils.ts [hunk 1]
```diff
@@ -1,3 +1,26 @@
+import { IconDefinition } from '@patternfly/react-icons/dist/esm/createIcon';
+
 export const removeTemplateWhitespace = (template: string): string => {
   return template.replace(/>(?:\s|\n)+</g, '><');
 };
+
```
[the remark this answers is not in the document]
Comment (logonoff) [reply frontend/packages/console-shared/src/components/markdown-extensions/utils.ts]: You make a good point, I've moved it to `@console/shared/src/utils/icon-utils`

**Comment by logonoff** [comment]
adding labels because this is debt:

/label px-approved
/label docs-approved

QE review:

/cc @yapei

File: frontend/packages/console-shared/src/components/markdown-extensions/inline-clipboard-extension.ts [hunk 2]
```diff
@@ -25,14 +26,14 @@ const useInlineCopyClipboardShowdownExtension = () => {
               <button class="pf-v5-c-button pf-m-plain" type="button" aria-label="${t(
                 'console-shared~Copy to clipboard',
               )}" ${MARKDOWN_COPY_BUTTON_ID}="${groupType}">
-                <i class="fas fa-copy" aria-hidden="true"></i>
```
[the remark this answers is not in the document]
Comment (logonoff) [reply frontend/packages/console-shared/src/components/markdown-extensions/inline-clipboard-extension.ts:28]: Unfortunately it's difficult to render a react component directly into a string, so instead, I used the `IconDefinition` object exported by every PF icon to generate an equivalent string containing the icon SVG markup

File: frontend/packages/console-shared/src/components/markdown-extensions/inline-execute-extension.ts [hunk 3]
```diff
@@ -36,8 +39,8 @@ const useInlineExecuteCommandShowdownExtension = () => {
                 <button class="pf-v5-c-button pf-m-plain ocs-markdown-execute-snippet__button" type="button" aria-label="${t(
                   'console-shared~Run in Web Terminal',
                 )}" ${MARKDOWN_EXECUTE_BUTTON_ID}="${groupType}">
-                  <i class="fas fa-play" aria-hidden="true"></i>
```
[the remark this answers is not in the document]
Comment (logonoff) [reply frontend/packages/console-shared/src/components/markdown-extensions/inline-execute-extension.ts:39]: Unfortunately it's difficult to render a react component directly into a string, so instead, I used the `IconDefinition` object exported by every PF icon to generate an equivalent string containing the icon SVG markup

File: frontend/packages/console-shared/src/components/markdown-extensions/inline-execute-extension.ts [hunk 4]
```diff
@@ -36,8 +39,8 @@ const useInlineExecuteCommandShowdownExtension = () => {
...
-                  <i class="fas fa-check" aria-hidden="true"></i>
```
[the remark this answers is not in the document]
Comment (logonoff) [reply frontend/packages/console-shared/src/components/markdown-extensions/inline-execute-extension.ts:40]: Unfortunately it's difficult to render a react component directly into a string, so instead, I used the `IconDefinition` object exported by every PF icon to generate an equivalent string containing the icon SVG markup

File: frontend/packages/console-shared/src/components/markdown-extensions/showdown-extension.scss [hunk 5]
```diff
@@ -15,20 +15,20 @@
 .ocs-markdown-execute-snippet {
 
   &__button {
-    & > i.fa-check {
```
[the remark this answers is not in the document]
Comment (logonoff) [reply frontend/packages/console-shared/src/components/markdown-extensions/showdown-extension.scss:18]: Unfortunately it's difficult to render a react component directly into a string, so instead, I used the `IconDefinition` object exported by every PF icon to generate an equivalent string containing the icon SVG markup

**Comment by yanpzhan** [comment]
Checked on cluster launched against the pr, check some pages on console including the screenshots in the pr description, there is not regression issue found.
/label qe-approved

**Pull request merged**